runtime/gc_base/MixedObjectAllocationModel.hpp

```diff
@@ -60,7 +61,9 @@ class MM_MixedObjectAllocationModel : public MM_JavaObjectAllocationModel
 		uintptr_t sizeInBytesRequired = clazz->totalInstanceSize + objectHeaderSize;
 
 #if defined (J9VM_GC_MODRON_COMPACTION) || defined (J9VM_GC_GENERATIONAL)
-		if (OMR_GC_ALLOCATE_OBJECT_HASHED == (OMR_GC_ALLOCATE_OBJECT_HASHED & allocateFlags)) {
+		if (OMR_GC_ALLOCATE_OBJECT_HASHED == (OMR_GC_ALLOCATE_OBJECT_HASHED & allocateFlags)
+			|| J9_ARE_ANY_BITS_SET(clazz->classFlags, J9ClassEnsureHashed)
```

Comment:
This still looks like hack to me. It is better to have `OMR_GC_ALLOCATE_OBJECT_HASHED` to be set in allocateFlags already

Reply:
@gacholio told me to check the flag (https://github.com/eclipse-openj9/openj9/pull/13338#issuecomment-903924138). This latest comment I'm not sure about, should `OMR_GC_ALLOCATE_OBJECT_HASHED` be set in `calculateRequiredSize`? https://github.com/eclipse-openj9/openj9/pull/13338#issuecomment-904006068 @gacholio could you please clarify

Reply:
`calculateRequiredSize()` is called from `MM_MixedObjectAllocationModel()` constructor only. It is called from `J9AllocateObject()` and J9AllocateObjectNoGC()`. I think this is correct level where we should check flag in class ant set it in allocation flags. Did I missed any path?

Reply:
Ah, forgot about the noGC path, so two places need the addition (which is 3 lines so I think we can stand the duplication), then no other changes in the GC are required.